src/server/scripts/BrokenIsles/MawOfSouls/boss_ymiron_the_fallen_king.cpp

```diff
@@ -0,0 +1,721 @@
+        GetTarget()->m_Events.AddEventAtOffset(new YmironSoulSiphonVisual(caster->GetGUID(), GetTarget(), travelSpeed), 1s);
+    }
+
+    void TriggerSuicide()
+    {
+        Creature* targetCreature = GetTarget()->ToCreature();
+        if (!targetCreature)
+            return;
+
+        targetCreature->CastSpell(nullptr, SPELL_SUICIDE, true);
+        targetCreature->DespawnOrUnsummon(20s);
+    }
+
+    void OnPeriodic(AuraEffect const* /*aurEff*/)
+    {
+        if (!roll_chance_f(10))
+            return;
+
+        TriggerSuicide();
+    }
+
+    void OnAfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
+            return;
+
+        TriggerSuicide();
+    }
+
+    void Register() override
+    {
+        AfterEffectApply += AuraEffectRemoveFn(spell_ymiron_the_fallen_king_soul_siphon::OnAfterApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_ymiron_the_fallen_king_soul_siphon::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_ymiron_the_fallen_king_soul_siphon::OnAfterRemove, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 167922 - Power
+class spell_ymiron_the_fallen_king_power_periodic : public AuraScript
+{
+    void OnPeriodic(AuraEffect const* aurEff)
+    {
+        int32 newPower = GetTarget()->GetPower(POWER_ENERGY) + (((aurEff->GetTickNumber() + 1) % 3 == 0) ? 6 : 7);
+        GetTarget()->SetPower(POWER_ENERGY, newPower);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_ymiron_the_fallen_king_power_periodic::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+class YmironBanePrecastVisual : public BasicEvent
+{
+public:
+    explicit YmironBanePrecastVisual(Unit* caster, int8 remainingVisualsToSpawn) : _caster(caster), _remainingVisualsToSpawn(remainingVisualsToSpawn) { }
+
+    static constexpr int8 BANE_VISUAL_NUM = 11;
+    static constexpr float BANE_VISUAL_TIME_OFFSET = 0.204545438289642f;
+    static constexpr float BANE_VISUAL_TIME_BASE = 0.75f;
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+    {
+        float dist = frand(10.0f, 35.0f);
+        float angle = frand(0.0f, 2.0f * float(M_PI));
+        Position dest = { _caster->GetPositionX() + dist * std::cos(angle), _caster->GetPositionY() + dist * std::sin(angle), _caster->GetPositionZ() };
+        float travelSpeed = BANE_VISUAL_TIME_BASE + (BANE_VISUAL_TIME_OFFSET * _remainingVisualsToSpawn);
+
+        _caster->SendPlaySpellVisual(dest, SPELL_VISUAL_BANE_PRECAST, 0, 0, travelSpeed, true);
+
+        _remainingVisualsToSpawn--;
+
+        if (_remainingVisualsToSpawn > 0)
+            _caster->m_Events.AddEventAtOffset(this, 200ms);
+
+        return false;
```

Comment:
false should only be returned if you are readding the event (otherwise its a memory leak)